File: Practice/unix 3, 37/Завдання 37/python/task37.py
#!/usr/bin/python3
import requests
import sys
import os
import re
from bs4 import BeautifulSoup
from collections import Counter

def deleteTags(html):
    soup = BeautifulSoup(html, "html.parser") 
    for data in soup(['style', 'script']):
         data.decompose() 
    return ' '.join(soup.stripped_strings)

if (sys.argv[1] == "--help"):
	print("Usage: task37.py Path")
	print("Select 100 most used words in HTML text  that you specify by path or URL")
	print("-h, --help show info about program")
	print("-t, --test start program with same parameters")

elif (sys.argv[1] == "--test"):
	os.system("./test37.bash")

elif (len(sys.argv) == 2):
	arg = sys.argv[1]
	if(os.path.isfile(arg)):
		text_from_file = open(sys.argv[1], "r").read()
		open('temporfile.txt', 'w').write(deleteTags(text_from_file))
	else:		
		page = requests.get(arg)
		open ('temporfile.txt', 'w').write(deleteTags(page.content))
	words = re.findall(r'\w+', open('temporfile.txt').read().lower())
	count = Counter(words).most_common(100)
	print(count)

File: Practice/unix 3, 37/Завдання 37/perl/task37.pl
#!/usr/bin/perl
if ($ARGV[0] eq "--help")
{
	print "Usage: task37.py Path\n";
	print "Select 100 most used words in HTML text  that you specify by path or URL\n";	
	print "-h, --help show info about program\n";	
	print "-t, --test start program with same parameters\n";	
}
elsif ($ARGV[0] eq "--test")
{
	system("./test37.bash");
}

else
{
	use strict;
	use warnings;
	use LWP::Simple qw(get);
	use HTML::Strip;
	use HTML::TokeParser::Simple;	
	my $outputfile = "outputfile.txt";
	my $arg = $ARGV[0];
 	if (-e $arg)
	{
                open file1, ">", $outputfile;
                my $p = HTML::TokeParser::Simple->new ($arg);
                while ( my $token = $p->get_token ){
                       print file1 $token->as_is if $token->is_text; }
                close (file1);
	}
	else
	{
		my $html = get $arg;
                my $hs = HTML::Strip->new();
                my $clean = $hs->parse($html);
		$hs->eof;		
		open file2, ">", $outputfile;
		print file2 $clean;
		close (file2);
	}
	my %hash;
        open my $file, '<', $outputfile;
        while (<$file>)	{ 
        	$hash{ lc $1 }++ while /(\w+)/g; }
        close $file;
        my $counter = 0;
        for my $word ( sort { $hash{$b} <=> $hash{$a} || $a cmp $b } keys %hash )
        {
		print "$counter. $word: ($hash{$word})\n" if $counter++ < 100 or !( $counter % 1000 );
        }

}
